### DISCUSSION TITLE
Disaster : yt-dlp dropping support for Python 3.8, our beloved QMPlay2 will no longer support windows 7 32 bits os ? or QMPlay2 will use alterantive yt-dlp made by Mr Nicolaasjan for 32 bits version ?

### DISCUSSION THREAD

**Discussion by trimechee** [zaps166/QMPlay2#768] [General]
Hi, I am sad to discover this Announcement: Dropping support for Python 3.8 ::

This will increase the minimum required Windows version to Windows 8; the yt-dlp release executables will no longer support Windows 7 or Windows Server 2008 R2.

Users on Windows 7 and Windows Server 2008 R2 will need to consider alternative options such as @nicolaasjan['s unofficial builds](https://github.com/nicolaasjan/yt-dlp/releases) if upgrading your operating system is not possible.

https://github.com/yt-dlp/yt-dlp/issues/10086

is our beloved QMPlay2  going to drop the 32-bit version for windows 7 os ?

we hope that windows7 os will continue to be supported please!
I use windows 7 and probably many people still have old laptops that are still functional, thank you!

**Comment by zaps166** [comment]
`yt-dlp` is external tool, it's out of scope.

I doubt the [unofficial builds](https://github.com/nicolaasjan/yt-dlp/releases) will be supported for long time.

---

You can manually replace the yt-dlp `exe` file in `%AppData%\QMPlay2` (rename unofficial exe if needed to replace existing file) directory with the unofficial version and it should follow the unofficial updates.

**Comment by trimechee** [comment]
Thank you very much for your important advices Mr @zaps166 and for to keep support of windows 7 os ! 💯 🥇 

We hope please to make the task easier for non expert users, we hope an out of the box installation by giving the choice and an option to users to install or not the unofficial version of yt-dlp 32 bits, maybe by inviting users to completely uninstall the old version of QMPlay2 and delete all installation folders to make a new easier installation of the unofficial version of yt-dlp 32 bits for users who want to install yt-dlp 32 bits after the end of the official version of yt-dlp , thank you very much !

Mr @nicolaasjan gives us very  precious advices and recommandations to compiling yt-dlp 32 bits and we hope  surely it will help to integrate unofficial yt-dlp to our beloved QMPlay2 media player !

https://github.com/mhogomchungu/media-downloader/issues/465#issuecomment-2423759207


 https://github.com/mhogomchungu/media-downloader/issues/465#issuecomment-2423761094 

https://github.com/adang1345/PythonWin7 

https://msfn.org/board/topic/183741-python-3813-for-windows-xp-sp3/page/2/#comment-1241782

https://msfn.org/board/topic/184368-who-here-has-a-youtube-dl-compile-for-winxp/page/9/

https://msfn.org/board/profile/414006-cmalex/ 

https://msfn.org/board/profile/347864-naaloh/

**Comment by nicolaasjan** [comment]
> You can manually replace the yt-dlp `exe` file in `%AppData%\QMPlay2` (rename unofficial exe if needed to replace existing file) directory with the unofficial version and it should follow the unofficial updates.

Unfortunately the unofficial updates can't be invoked with `-U`.
They have to be manually updated from [Releases](https://github.com/nicolaasjan/yt-dlp/releases).

> I doubt the unofficial builds will be supported for long time.

I will do my best, but I'm not so young any more...
Besides that, I have to build the Windows XP version with a lower PyInstaller version (5.13.2) and I don't know how long that will work.

**Comment by trimechee** [comment]
Thanks again Mr @nicolaasjan  for achieving the rare technical feat and having the immense generosity to add a 32-bit Windows 7 version while you are using Windows XP! and thanks for Mr @zaps166 for the support of 32 bits os !

If doing these versions can make you feel bad or cause any problems Mr Nicolaasjan,  then please Mr @nicolaasjan , please don't get tired anymore,  we will use online sites to download youtube videos :)

I installed 3D Youtube Downloader which is compatible with Windows 7 32-bit, and 3D Youtube Downloader seems not to use yt-dlp and seems to use its own original technique to download YouTube, maybe it will be the ultimate solution if there is no more 32-bit yt-dlp Windows 7 :)

I also discovered MassTube which is compatible with Windows 7 os but I have never tried this software which is apparently freemium:

http://www.masstube.cl/

**Comment by nicolaasjan** [comment]
> and having the immense generosity to add a 32-bit Windows 7 version while you are using Windows XP!

(offtopic)

Actually I'm using Linux Mint 21.3 64bit on an old Intel Core i5-4670 with Intel Xeon E3-1200 v3/4th Gen Core Processor Integrated Graphics and 16GB RAM

The Windows machines are all virtual, but run fast on SSD. :)
The Windows XP machine starts up to a usable desktop in ~10seconds.

![Schermafdruk op 2024-10-19 20-23-10](https://github.com/user-attachments/assets/56de8048-c553-44f5-9bd5-c0e6826c527b)

**Comment by trimechee** [comment]
Interesting and very informative ! Thanks !  yes  virtual machines looks great and speed  and  i think old legacy os are better than modern os, just the obstacle course that often fails with windows 11 updates that cancel our modifications, for example disabling the anti virus antimalware service which can be annoying, in my windows 7 os, it is very easy to disable the antivirus, but on windows 11, it is a nightmare.....
between activating safe mode, modifying registry values, group policy, having pro version of windoss to have group policy.....

but fortunately I found 3 rare software that allow us to easily disable the windows 11 antivirus and I hope to test them if I get a new laptop windos 11 in sha Allah!

**Comment by zaps166** [comment]
> but fortunately I found 3 rare software that allow us to easily disable the windows 11 antivirus and I hope to test them if I get a new laptop windos 11 in sha Allah!

Try to run safe mode and do it in regedit with instruction from network. Better to avoid any software :smile: Also don't forget to disable some telemetry.

**Comment by trimechee** [comment]
Thank you for your advices and recommandations :) 💯 

yes in my windows 10 pro laptop which crashed, no problem, I was able to disable the antivirus but sometimes the registry editor value refuses to change.....apparently there is a conflict between the group policy and the registry editor maybe because I did not activate the safe mode...

in my sister's original windows 11 home edition computer, I disabled the antivirus but a new update reactivated the antimalware service, I installed policy plus to try to access the group policy and disable windows defender but the windows defender section no longer exists in the windows 11 group policy!!

https://github.com/Fleex255/PolicyPlus

anti virus anti malware allows windows to completely control our computer, block windows activator crack for example....and many have expressed fear that the TPM requirement in windows 11 which is a kind of DRM and windows can now block our access to websites if windows decides....

so if I get a new windows 11 laptop in sha Allah , despite I have done immense research and I have a hundred pages how to disable the antivirus, I will now prioritize peace of mind and avoid manipulating the sensitive registry editor which can cause instability and use these 3 hyper rare gamer changer software including 2 are open source, I spoke to one of the developers to ask him if I can disable the antivirus on windows 11 home because I do not have the pro edition and he reassured me that yes and that his software does not use group policy to disable windws defender but use the bios, which allows me in sha Allah to buy a laptop  entry level because disabling the anti virus anti malware service boosts performance and no need for an expensive cpu or noisy fan, in addition with secure dns like quad9, dns0 zero, cira dns, control d dns, cleanbrowsing, we can protect ourselves from viruses without anti virus:

https://winaero.com/winaero-tweaker/

https://github.com/hellzerg/optimizer

https://github.com/semazurek/ET-Optimizer


i read disable telemetry can break some services, so regarding telemetry, I no longer disable telemetry, stability takes precedence over privacy, the same goes for ad blocker dns, some google analytics domains are necessary for the normal functioning of websites, banking services.....
that's why I no longer use ad blocker dns and I disable ublock origin when I use gmail, sensitive sites and I use dns that only block malware viruses....

**Comment by trimechee** [comment]
Mr @zaps166 , here interesting new post about automatic update of the new unofficial yt-dlp 32 bits : https://github.com/mhogomchungu/media-downloader/issues/465#issuecomment-2425299419

**Comment by trimechee** [comment]
Great annoncement from the generous Mr Bashonly ! : 

https://github.com/mhogomchungu/media-downloader/issues/465#issuecomment-2425481297

**Comment by trimechee** [comment]
Mr @nicolaasjan , thanks for the news versions ! 💯 🥇 

I have some suggestions please to lighten your heavy compilation workload: I suggest that you no longer compile the with in-between releases i.e. no need to compile the beta / alpha versions of yt-dlp, and just compile the stable version of yt-dlp which comes out every 2 months, and if the workload remains very heavy then it is enough to compile 2 or 3 releases per year if there are important fixes for youtube because I think people use yt-dlp mainly for youtube, if the stable version of yt-dlp does not contain changes for youtube, then no need to compile the new stable version of yt-dlp because what matters is just the youtube changelogs .....

and I suggest you Mr @ to support your essential life saver projects unique in the world which save millions of machines and the ecology and our wallets, I suggest highlighting a "donate" button to encourage users to do donations to support your wonderful projects, or add sponsor button on the unofficial yt dlp github page, and I also suggest to Mr @zaps166  to put a "donation" button at the launch of QMPlay2 to encourage people to support the project, thank you :)

**Comment by trimechee** [comment]
Awesome  new titanic vital important announcement from Mr Nicolaasjan !!! : 

https://github.com/yt-dlp/yt-dlp/issues/10086#issuecomment-2428755973

**Comment by trimechee** [comment]
Mr @zaps166 , so we should add manually the new yt-dlp for windows 7 os to QMPlay2 ou we should wait the new version of QMPlay2 32 bits which will add and update automatically the new version of yt-dlp for windows 7 os 32 bits ?

**Reply by zaps166** [reply]
New version 24.12.06 allows you to set path for custom yt-dlp/youtube-dl executable. It's also possible to automatically update it if it updates the same way as modern yt-dlp.

**Comment by trimechee** [comment]
Awesome !!! Huge thanks and gratitude for all these vtal life saver features Mr @zaps166  !! 💯 🥇
